File: app/src/main/java/com/simarjot/bookwala/cat/SchoolCatActivity.java
package com.simarjot.bookwala.cat;

import androidx.appcompat.app.AppCompatActivity;

import android.content.SharedPreferences;
import android.net.Uri;
import android.os.Bundle;
import android.widget.TextView;

import com.makeramen.roundedimageview.RoundedImageView;
import com.simarjot.bookwala.R;
import com.simarjot.bookwala.helpers.Helper;
import com.warkiz.widget.IndicatorSeekBar;
import com.warkiz.widget.OnSeekChangeListener;
import com.warkiz.widget.SeekParams;

public class SchoolCatActivity extends AppCompatActivity {
    //widgets
    private IndicatorSeekBar mClassSeekBar;
    private TextView mClassTV;
    private RoundedImageView mCoverImage;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_school_cat);
        mCoverImage = findViewById(R.id.cover_image);
        mClassTV = findViewById(R.id.text);
        mClassSeekBar = findViewById(R.id.class_seek_bar);

        mClassSeekBar.setOnSeekChangeListener(new OnSeekChangeListener() {
            @Override
            public void onSeeking(SeekParams seekParams) {

            }

            @Override
            public void onStartTrackingTouch(IndicatorSeekBar seekBar) {

            }

            @Override
            public void onStopTrackingTouch(IndicatorSeekBar seekBar) {
                int progress = seekBar.getProgress();
                if(progress==1){
                    mClassTV.setText(progress + "st");
                }else if(progress==2){
                    mClassTV.setText(progress + "nd");
                }else if(progress==3){
                    mClassTV.setText(progress + "rd");
                }else if(progress>3 && progress<13){
                    mClassTV.setText(progress + "th");
                }
            }
        });
        SharedPreferences prefs = getSharedPreferences(Helper.BOOK_IMAGE_PREFS, MODE_PRIVATE);
        String coverImageUrl = prefs.getString(Helper.COVER_IMAGE, null);
        mCoverImage.setImageURI(Uri.parse(coverImageUrl));
    }
}

File: app/src/main/java/com/simarjot/bookwala/cat/SchoolCatActivity.kt
package com.simarjot.bookwala.cat

import android.content.Context
import android.net.Uri
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.text.Editable
import android.text.TextWatcher
import android.view.View
import android.widget.EditText
import com.simarjot.bookwala.R
import com.simarjot.bookwala.helpers.Helper
import com.skydoves.powerspinner.createPowerSpinnerView
import kotlinx.android.synthetic.main.activity_school_cat.*

class SchoolCatActivity : AppCompatActivity() {
    var doneClicked = false

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_school_cat)
        val prefs = getSharedPreferences(Helper.BOOK_IMAGE_PREFS, Context.MODE_PRIVATE)
        val coverImageUri = prefs.getString(Helper.COVER_IMAGE, null)
        cover_image.setImageURI(Uri.parse(coverImageUri))

        title_et.addTextChangedListener(object : TextWatcher {
            override fun afterTextChanged(p0: Editable?) {
            }

            override fun beforeTextChanged(p0: CharSequence?, p1: Int, p2: Int, p3: Int) {
            }

            override fun onTextChanged(text: CharSequence?, p1: Int, p2: Int, p3: Int) {
                if(doneClicked){
                    if(text.isNullOrBlank()){
                        setError(title_et, "Title is mandatory")
                    }else{
                        removeError(title_et, "Book Title")
                    }
                }
            }
        })

        subject_et.addTextChangedListener(object: TextWatcher{
            override fun afterTextChanged(s: Editable?) {
            }

            override fun beforeTextChanged(s: CharSequence?, start: Int, count: Int, after: Int) {
            }

            override fun onTextChanged(text: CharSequence?, start: Int, before: Int, count: Int) {
                if(doneClicked){
                    if(text.isNullOrBlank()){
                        setError(subject_et, "Subject is mandatory")
                    }else{
                        removeError(subject_et, "Subject")
                    }
                }
            }
        })

        done_btn.setOnClickListener{
            doneClicked = true
            if(title_et.text.isNullOrBlank()){
                setError(title_et, "Title is mandatory")
            }

            if(subject_et.text.isNullOrBlank()){
                setError(subject_et, "Subject is mandatory")
            }

        }
    }

    private fun setError(view: EditText, msg: String){
        view.setBackgroundResource(R.drawable.background_error)
        view.hint = msg
    }
    private fun removeError(view: EditText, msg: String){
        view.setBackgroundResource(R.drawable.rounded_edit_text_white)
        view.hint = msg
    }
}
